pages/docs/tutorials/native/using-intellij-idea.md

```diff
@@ -1,50 +1,126 @@
+
+2. Change the code of `main()`:
+
+   * Use the [`readLine()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/read-line.html) function to read the input value to the `name` variable.
+   * Check the input value with the [safe call operator `?.`](https://kotlinlang.org/docs/reference/null-safety.html#safe-calls).
+   * Use the [`replace()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/replace.html) function to eliminate the whitespaces in the name.
+   * Use the scope function [`let`](https://kotlinlang.org/docs/reference/scope-functions.html#let) to execute function within the object context. 
+   * Use a [string template](https://kotlinlang.org/docs/reference/basic-types.html#string-templates) to insert your name into the string by adding a dollar sign `$` and enclosing it in curly braces – `${name?.length}`.
+   * Report about the null value using the [`error()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/error.html) function after the [Elvis operator `?:`](https://kotlinlang.org/docs/reference/null-safety.html#elvis-operator). 
+
+   <div class="sample" markdown="1" theme="idea" mode="kotlin" data-highlight-only>
+
+   ```kotlin
+   fun main() {
+       println("Hello, enter your name:")
+       val name = readLine()
+       name?.replace(" ", "")?.let {
+           println("Your name contains ${it.length} letters")
+       } ?: error("Error while reading input from the terminal: the value can't be null.")
+   }
+   ```
+
+   </div>
+
+3. Save the changes and run the application.
+
+   IntelliJ IDEA opens the **Run** tab and shows the output.
+
+4. Enter your name and enjoy the result:
+
+   ![Application output]({{ url_for('tutorial_img', filename='native/using-intellij-idea/native-new-project-intellij-5.png') }})
+
+
+### Count unique letters in your name
+
+1. Open the file `main.kt` in `src/<your_app_name>Main/kotlin`.
+
+2. Count the distinct letters in your name:
+
+   * Convert the name to lower case using the [`toLowerCase()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/to-lower-case.html) function.
+   * Convert the input string to a collection of characters using the [`toList()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/to-list.html) function.
+   * Select only distinct characters in your name using the [`distinct()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/distinct.html) function.
+   * Count distinct characters using the [`count()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/count.html) function.
+
+   <div class="sample" markdown="1" theme="idea" mode="kotlin" data-highlight-only>
+
+   ```kotlin
+   fun main() {
+       println("Hello, enter your name:")
+       val name = readLine()
+       name?.replace(" ", "")?.let {
+           println("Your name contains ${it.length} letters")
+           println("Your name contains ${it.toLowerCase().toList().distinct().count()} unique letters")
```

Comment:
Looks a bit long for such small example. May be it would be better to do extension function
```
fun String.countDistinctCharacters() = toLowerCase().toList().distinct().count()
fun main() {
...
println("Your name contains ${it.countDistinctCharacters()} unique letters")
}
```

Or extension property

```
val String.distinctCharactersNumber:Int
    get() = toLowerCase().toList().distinct().count()
```

Reply:
It makes the sample looking too complex, doesn't it? I'd leave the current version instead if you don't mind.

Reply:
The main problem is that the second print is too long. If I open staging link, this string even doesn't fit a code block width.
Again I amn't aware of the purpose of this tutorial. If we want to show syntax sugar of Kotlin to write readable code extension function is better. If this tutorial is for people without any background at least we can make a variable `val distinctCharacters = it.toLowerCase().toList().distinct().count()`. It would be more readable on web page.

Reply:
Ok, I got it.
I took an example with an extension function – I guess it looks more clear because we use only functions in that sample.